lib/contentful/base_resource.rb

```diff
@@ -0,0 +1,87 @@
+require_relative 'support'
+
+module Contentful
+  # Base definition of a Contentful Resource containing Sys properties
+  class BaseResource
+    attr_reader :raw, :default_locale, :sys
+
+    def initialize(item, configuration = {}, _localized = false, _includes = [], depth = 0)
+      @raw = item
+      @default_locale = configuration[:default_locale]
```

Comment:
It's a copy of the configuration object sent to the resource builder, can be seen here: https://github.com/contentful/contentful.rb/pull/113/files#diff-ccad056e91bbffbe74262ffa2994a804R277

It's non-essential and internal, if you want to create your own instances it's not required to send anything there.